src/utils/tree_node.C

```diff
@@ -457,25 +460,30 @@ unsigned int TreeNode<N>::n_active_bins() const
 
 
 template <unsigned int N>
-const Elem* TreeNode<N>::find_element(const Point& p, const std::set<subdomain_id_type> *allowed_subdomains) const
+const Elem*
+TreeNode<N>::find_element
+(const Point& p,
+ const std::set<subdomain_id_type> *allowed_subdomains,
+ Real relative_tol) const
 {
   if (this->active())
     {
       // Only check our children if the point is in our bounding box
       // or if the node contains infinite elements
-      if (this->bounds_point(p) || this->contains_ifems)
+      if (this->bounds_point(p, relative_tol) || this->contains_ifems)
```

Comment:
Ha!  Agreed.  But I'd like to get things completely working first, then get them factored more nicely second.

Dropping bounding box tolerances back to sanity didn't change any of my results... which means that right now I can only trigger failures on a multi-million-element mesh, which makes ongoing debugging tricky.